devito/passes/iet/mpi.py

```diff
@@ -182,7 +183,12 @@ def rule2(dep, hs, loc_indices):
             # within the same Conditional, otherwise we would break the control
             # flow semantics
             if cond_mapper.get(hs0) != cond_mapper.get(hs1):
-                continue
+                cond = cond_mapper.get(hs1).symmetric_difference(cond_mapper.get(hs0))
```

Comment:
would it be neater to de-nest it by re-writing it as (pseudocode):

```
cond = cond_mapper.get(hs1).symmetric_difference(cond_mapper.get(hs0))
if len(cond) > 1:
    <explanation>
    continue
elif len(cond) == 1 and isinstance(cond.pop().condition, GuardFactorEq):
    <explanation>
    continue
```

Reply:
cleaner to just filter when making the cond_mapper